### PR TITLE
Use default namespace from useTranslation hook if supplied

### PR DESCRIPTION
**Pull request by heivo**

### CODE REVIEW

**Comment by heivo** [comment]
This is a fix for issue #161

File: src/lexers/javascript-lexer.js [hunk 1]
```diff
@@ -74,9 +74,17 @@ export default class JavascriptLexer extends BaseLexer {
         }
       }
 
+      if (this.defaultNamespace) {
+        entry.namespace = this.defaultNamespace;
```
Comment (karellm) [inline src/lexers/javascript-lexer.js]: For consistency purposes, could you remove the `;`?

Comment (heivo) [reply src/lexers/javascript-lexer.js]: sure, done

**Comment by heivo** [comment]
I added a feature to evaluate namespace overrides in the t-function options (see docs here: https://www.i18next.com/translation-function/essentials).

**Comment by hschmied** [comment]
@heivo if you could just help me out for a minute, I looked at the commit and see that your code checks if a namespace is given in useTranslation 'string' as well as an array of namespaces, in which case the first one is considered default. (lines 80-82 in your javascript-lexer.js commit)

we are using 1.0.6 of the parser, but for whatever reason, when using a namespace array when initializing 't', the parser is not parsing correctly keys without given namespace (falling back on default).

```
const {t, i18n} = useTranslation('localization', {useSuspense: true});
...
<Text>{t('test_fig')}</Text>
```

... correctly puts it into localization.json ...

```
const {t, i18n} = useTranslation(['localization', 'app', 'common'], {useSuspense: true});
...
<Text>{t('test_fig')}</Text>
```

... does not. instead it writes it into translation.json

---

Btw -- the test included with the commit only checks for useTranslate with namespace-string, but not array.

Thanks & best regards!
-H

**Comment by heivo** [comment]
@hschmied I think the problem is more related to lines 90-92: when an array is supplied as the first argument the namespace cannot be read from `node.arguments[0].text` but from `node.arguments[0].elements[0].text`.

Something like this should work:
```js
    if (
      node.expression.escapedText === "useTranslation" &&
      node.arguments.length
    ) {
      const { text, elements } = node.arguments[0];
      if (text) {
        this.defaultNamespace = text;
      } else if (elements && elements.length) {
        this.defaultNamespace = node.arguments[0].elements[0].text;
      }
    }
```

**Comment by hschmied** [comment]
@heivo jep... that did it and works beautifully! How do you want to proceed to get this change in? I mean you fixed it and should get the credit. :)

**Comment by karellm** [comment]
This should be fixed in `1.0.7`.

**Pull request merged**